src/evaluate/evaluator/question_answering.py

```diff
@@ -136,89 +175,18 @@ def compute(
-            label_column (`str`, defaults to `"answers"`):
-                the name of the column containing the answers in the dataset specified by `data`.
-            squad_v2_format (`bool`, *optional*, defaults to `None`):
-                whether the dataset follows the format of squad_v2 dataset where a question may have no answer in the context. If this parameter is not provided,
-                the format will be automatically inferred.
-        Return:
-            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
-            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
-            containing the score, the confidence interval and the standard error calculated for each metric key.
-
-        Examples:
-        ```python
-        >>> from evaluate import evaluator
-        >>> from datasets import load_dataset
-        >>> task_evaluator = evaluator("question-answering")
-        >>> data = load_dataset("squad", split="validation[:2]")
-        >>> results = task_evaluator.compute(
-        >>>     model_or_pipeline="sshleifer/tiny-distilbert-base-cased-distilled-squad",
-        >>>     data=data,
-        >>>     metric="squad",
-        >>> )
-        ```
-
-        <Tip>
-
-        Datasets where the answer may be missing in the context are supported, for example SQuAD v2 dataset. In this case, it is safer to pass `squad_v2_format=True` to
-        the compute() call.
-
-        </Tip>
-
-        ```python
-        >>> from evaluate import evaluator
-        >>> from datasets import load_dataset
-        >>> task_evaluator = evaluator("question-answering")
-        >>> data = load_dataset("squad_v2", split="validation[:2]")
-        >>> results = task_evaluator.compute(
-        >>>     model_or_pipeline="mrm8488/bert-tiny-finetuned-squadv2",
-        >>>     data=data,
-        >>>     metric="squad_v2",
-        >>>     squad_v2_format=True,
-        >>> )
-        ```
+        question_column (`str`, defaults to `"question"`):
+            the name of the column containing the question in the dataset specified by `data`.
+        context_column (`str`, defaults to `"context"`):
+            the name of the column containing the context in the dataset specified by `data`.
+        id_column (`str`, defaults to `"id"`):
+            the name of the column cointaing the identification field of the question and answer pair in the
+            dataset specified by `data`.
+        label_column (`str`, defaults to `"answers"`):
+            the name of the column containing the answers in the dataset specified by `data`.
+        squad_v2_format (`bool`, *optional*, defaults to `None`):
+            whether the dataset follows the format of squad_v2 dataset where a question may have no answer in the context. If this parameter is not provided,
+            the format will be automatically inferred.
```

Comment:
I'd rename this param (and similar util functions below) to something more general e.g. `answer_optional`, not having answer in the context is a thing in other datasets as well, e.g. in long form QA. Such name is also more explainable for people unfamiliar with `squad_v2`

Reply:
since we already released it that would be a breaking change so I'd avoid it. maybe we can expand the description instead?

Reply:
Done!